bridget/eddy/footprint.py

```diff
@@ -45,4 +282,78 @@ def footprint(t_air, a, p, u, cov_uw, cov_vw, cov_wt, var_v, z = 2.0, method = "
         air pressure [hPa].
     """
 
-    raise NotImplementedError
+    u_star = u_star_calc(cov_uw, cov_vw)
+
+    e_w = sat_vapor_pressure_calc(p, t_air)
+
+    rh = relative_humidity_calc(a, t_air, e_w)
+
+    e_a = act_vapor_pressure_calc(rh, e_w)
+
+    rho = air_density_calc(p, e_a, t_air)
+
+    c_p = specific_heat_cap_calc(e_a, p)
+
+    hts = sensible_heat_calc(rho, c_p, cov_wt)
+    #results for hts are pretty much in compliance with the results in the TK2-file
+    #but many numbers are -9999 in the TK2-file -> when is hts not defined??
+
+    L = obukhov_length_calc(rho, c_p, u_star, t_air, hts)
+
+
+        #http://www.shodor.org/os411/courses/_master/tools/calculators/virtualtemp/tv1calc.html
+        ###auch mal in Knöfel, p. 36, Energiebilanzmodellierung schauen, hier alle Formeln!
+        #z/L is differing from the values in the TK2-file --> look at the calculation aggregation
+
+    #x) specific humidity and virtual temperature
+    q = 0.622 * (e_a/(p - 0.378 * e_a))     #[kg/kg]    Foken, 2017, p. 49
+    t_virt = (1 + 0.61 * q) * t_air         #[°C]       Foken, 2017, p. 37
+
+
+
+
+    ### core-footprint-calculation starts here
+
+    #f.1) calculate phi_c and phi_m (stability functions) [-], Kormann_Meixner gives no calculation if L == 0
+    for i in L:
+        if (i > 0):
+            phi_c = 1 + 5 * z/L
+            phi_m = 1 + 5 * z/L
+        elif (i < 0):
+            phi_c = (1 - 16 * z/L)**(-1/2)
+            phi_m = (1 - 16 * z/L)**(-1/4)
+
+    #f.2) calculate m (exponent of the wind velocity power law) and n (exponent of the eddy diffusivity power law) [-]
+    k = 0.4    # von Karman constant
+
+    m = (u_star * phi_m) / (k * u)
+
+    for i in L:
+        if (i > 0):
+            n = 1 / (1 + 5 * z/L)
+        elif (i < 0):
+            n = (1 - 24 * z/L) / (1- 16 * z/L)
+
+    #f.3) calculate r (shape factor) [-]
+    r = 2 + m - n
+
+    #f.4) calculate eddy diffusivity K [m2/s]
+    K = (k * u_star * z) / (phi_c * z/L)
+
+    #f.5) calculate U (constant in power law profile of the wind velocity) [m^(1-m)/s] and kappa (constant in power law profile of the eddy diffusivity) [m^(2-n/m)] (dimensions not realy working here)
+    U = u / z**m
+    kappa = K / z**n
+
+    #f.6) calculate xi (flux length scale) [m]
+    xi = (U * z**r) / (r**2 * kappa)
+
+
+    #f.7) calculate the flux footprint [1/m2]
+    mu = (1 + m) / r
+
+
+
+
+
+
+    pass
```

Comment:
I think a return statement is missing, right?

Reply:
Yes, I´m currently working on the footprint function, but I decided to do that in another branch.

Reply:
Regarding the constants: I think these are often empirical solutions for the calculation of the variables. 
But you are right, I also noticed that some numbers or even formulas are varying in the literature. It was often difficult to find a usable formula at all so I was happy to find something that at least worked.

Maybe we could create a new issue for that as it is some work to go through the literature again and check for that.

Reply:
> Maybe we could create a new issue for that as it is some work to go through the literature again and check for that.

I don't think that the two of us need to decide which constant etc. to use. Let's just expose the contants as parameters with default valule, like you did with the k parameter. I we don't do that in this PR, we should open an issue to not forget about it

> Yes, I´m currently working on the footprint function, but I decided to do that in another branch.
 Perfect. I'll review your changes again.